### PR TITLE
[rl] Add trainer entropy metric

### PR DESCRIPTION
**Pull request by acisseJZhong**
## Summary
Add a trainer-policy **entropy** metric to watch for entropy -> reward collapse during RL (motivated by checking whether MoE / Qwen3-30B-A3B runs collapse), and surface the existing train/inference **KL** signal.

## Why
Entropy collapse is the leading indicator of reward collapse in GRPO/DAPO: the policy sharpens to near-deterministic -> within-group reward variance -> 0 -> zero advantage -> no gradient. Entropy + KL on the dashboard catch this early.

## What
- `compute_logprobs(..., return_entropy=True)` returns per-token entropy alongside logprobs from a **single vocab gather + single fp32 upcast** (no extra copy); default call is unchanged (returns logprobs).
- DAPO logs **`trainer/entropy/mean`** (mean `H(p)` over response tokens); documents `bit_wise/logprob_diff/mean` as the k1 estimate of `-KL(q||p)` (`-1x` verl's `ppo_kl`).
- Rename `train/*` -> `trainer/*` (`grad_norm`, `lr`, `policy_version`); add entropy to the console keys.

## Test plan
- Ran locally (`rl_grpo_qwen3_1_7b` search_r1 + alphabet_sort); `trainer/entropy/mean` logs to W&B with the expected sharpen-then-plateau.

### CODE REVIEW

File: torchtitan/components/loss.py [hunk 1]
```diff
@@ -415,6 +419,44 @@ def compute_logprobs(
...
+    - entropy ``H(p) = logsumexp(logits) - sum(softmax(logits) * logits)`` is a
+      metric only, so it is computed under ``no_grad``: it never contributes
+      gradient and must not build an autograd graph over the [B, L, V] softmax.
+
+    Both outputs are ``[batch, seq_len]``.
+    """
+    logits = _gather_vocab_logits(logits)
+    logprobs = _logprobs_from_full_vocab(logits, labels)
+    with torch.no_grad():
+        logits_fp32 = logits.float()
```
Comment (tianyu-l) [inline torchtitan/components/loss.py]: This way you'd have two `.float()` conversion, one in logprobs, one for entropy. Have you thought about that?

File: torchtitan/components/loss.py [hunk 2]
```diff
@@ -415,6 +419,44 @@ def compute_logprobs(
     ).reshape(B, L)
 
 
+def compute_logprobs(
```
Comment (tianyu-l) [inline torchtitan/components/loss.py]: Tow much value is it to keep this standalone? There are two other options so that we don't have to introduce many subfunctions:
- remove this, and only use the combined function
- having a flag e.g. `return_entropy` to if-else

curious what's the tradeoff in your mind?

File: torchtitan/components/loss.py (hunk 1 above)
Comment (acisseJZhong) [reply torchtitan/components/loss.py]: good point, we shouldn't keep two copies of float32 logits, we can share one.

File: torchtitan/components/loss.py (hunk 2 above)
Comment (acisseJZhong) [reply torchtitan/components/loss.py]: compute_logprobs is only used in some test scripts that we have under experimental/rl. I think combining these two functions(second option) looks better.

File: torchtitan/components/loss.py [hunk 3]
```diff
@@ -372,12 +372,24 @@ def __init__(self, config: Config, *, compile_config: CompileConfig | None = Non
 def compute_logprobs(
     logits: torch.Tensor,
     labels: torch.Tensor,
-) -> torch.Tensor:
-    """Compute per-position logprobs from logits and labels.
+    *,
+    return_entropy: bool = False,
+) -> torch.Tensor | tuple[torch.Tensor, torch.Tensor]:
```
Comment (tianyu-l) [inline torchtitan/components/loss.py:377]: The other option is to return `logprobs, None` when `return_entropy=False`. Not sure which is better.

Update: OK claude prefers o/w, so no need to change

**Review by tianyu-l — APPROVED** [review]

**Comment by tianyu-l** [comment]
@claude review this PR, including my comments
